### PR TITLE
bazel: add fixed output test case

### PR DESCRIPTION
**Pull request by timokau**
###### Motivation for this change

Feels like a horrible hack, but it should make sure that downstream
hashes are preserved between bazel versions.

The hash changed in the 0.24 -> 0.26 update.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by Profpatsch** [comment]
I’ll take a look on Monday

**Comment by timokau** [comment]
Looks like my hack is not actually necessary, since `bazel` is already in `nativeBuildInputs` of the fixed output derivation and therefore any change to bazel will re-trigger the fixed  output derivation build.

I seem to remember that I had some problems with `fetchpatch` not getting re-evaluated even though `patchutils` had changed, but apparently that is not the case here for some reason :thinking: 

Anyway, I'll remove the hack.

**Comment by Profpatsch** [comment]
Can you put it in a sub-attributeset? Then one can do `nix-build -A bazel.tests.downstreamPackages` or something.

**Comment by timokau** [comment]
Done

File: pkgs/development/tools/build-managers/bazel/default.nix [hunk 1]
```diff
@@ -173,6 +175,12 @@ stdenv.mkDerivation rec {
 
       pythonBinPathWithNixHacks = callPackage ./python-bin-path-test.nix{ inherit runLocal bazelTest; bazel = bazelWithNixHacks; };
       bashToolsWithNixHacks = callPackage ./bash-tools-test.nix { inherit runLocal bazelTest; bazel = bazelWithNixHacks; };
+
+      # downstream packages using buildBazelPackage (this tests that the fixed output derivation hashes have not changed)
+      downstream = {
+        dm-sonnet = python.pkgs.dm-sonnet;
+        bazel-watcher = bazel-watcher;
```
Comment (kalbasit) [inline pkgs/development/tools/build-managers/bazel/default.nix]:
```suggestion
        inherit bazel-watcher;
```

**Review by kalbasit — DISMISSED** [review]

**Comment by kalbasit** [comment]
@GrahamcOfBorg build bazel.tests.downstreamPackages

**Comment by timokau** [comment]
I just noticed that the `downstream` tests won't actually be run when running `nix build -f. bazel.tests`. Is that what you intended @Profpatsch?

**Comment by Profpatsch** [comment]
> I just noticed that the `downstream` tests won't actually be run when running `nix build -f. bazel.tests`. Is that what you intended @Profpatsch?

Oh, hm. Maybe we need to add `recurseIntoAttrs`.

**Comment by Profpatsch** [comment]
@GrahamcOfBorg build bazel.tests

**Comment by Profpatsch** [comment]
@GrahamcOfBorg build bazel.tests

**Comment by Profpatsch** [comment]
`dm-sonnet` is failing on linux.
`bazel-watcher` is failing on darwin (as noted in https://github.com/NixOS/nixpkgs/issues/65268).

@kalbasit can you take a look at `bazel-watcher`? @timokau you are maintaining `dm-sonnet`, can you look into it?

**Comment by kalbasit** [comment]
@Profpatsch bazel-watcher fix is in #65308.

**Comment by timokau** [comment]
> you are maintaining dm-sonnet, can you look into it?

Not currently. If nobody else steps up, it'll have to wait for a week or two.

**Comment by Profpatsch** [comment]
We should also add `bazel-buildtools` to the downstream tests.

> Not currently. If nobody else steps up, it'll have to wait for a week or two.

I’ll remove it from the dependency test for now and open a new issue.

**Comment by Profpatsch** [comment]
Rebased on master & disabled dm-sonnet

**Comment by Profpatsch** [comment]
And we’re live.

**Comment by Profpatsch** [comment]
Tests went through, autosquashing and merging.

**Pull request merged**